packages/flutter/lib/src/material/carousel.dart

```diff
@@ -146,12 +199,37 @@ class Carousel extends StatefulWidget {
   /// Defaults to false.
   final bool reverse;
 
+  /// Whether we allow the "squished" item to expand to the max size.
+  ///
+  /// If this is false, the layout of the carousel doesn't change. This is especially
+  /// useful when the carousel has a centered-hero layout in which the max item
+  /// is in the middle and at least one small item on each side, the first and
+  /// the last item cannot expand to the max size. If this is true, there will
+  /// be some space before/after the first/last item coming so every items have
+  /// a chance to be fully expanded.
+  ///
+  /// Defaults to true on hero layout and false on multi-browse layout.
+  final bool allowFullyExpand;
+
   /// Called when one of the [children] is tapped.
   final ValueChanged<int>? onTap;
 
   /// The extent the children are forced to have in the main axis.
+  ///
+  /// This is required for uncontained layout. For [Carousel.weighted], this is null.
   final double? itemExtent;
 
+  /// The weights that each visible child should occupy the viewport.
+  ///
+  /// The length of [layoutWeights] means how many items we want to lay out on
+  /// the viewport. For example, setting [layoutWeights] to `<int>[3,2,1]` means
+  /// there are 3 carousel items and their extents are 3/6, 2/6 and 1/6 of the
+  /// viewport extent.
+  ///
+  /// This is a required property when [Carousel.multibrowse] or [Carousel.hero]
```

Comment:
Neither of these constructors exist yet? Also see my remark above.

Reply:
@navaronbracke I think this is because this is Part 2 of 2. There is another PR that is part 1.

Reply:
I also looked through that one, but https://github.com/flutter/flutter/pull/148094/files#diff-5fabc52f74ea4158bcf844529f5daacf7026368d8cc8915e086d332f6208ab42R53 defines only the unnamed constructor for the carousel. And this PR adds `weighted`, but no others (yet?).

Reply:
Thanks for catching this! This doc was a little out of date. I created `Carousel.multibrowse` and `Carousel.hero` but later noticed they just have almost the same parameters, so I just merged them together by calling it `Carousel.weighted`. Just updated:)